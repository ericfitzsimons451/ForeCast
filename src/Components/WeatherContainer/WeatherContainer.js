import React, { Component } from 'react'
import './WeatherContainer.scss'
import City from '../City/City'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'

export class WeatherContainer extends Component {

	render() {
		return (
			<div className='weather-container'>
				<City key={this.props.currentWeather.city_name} currentWeather={this.props.currentWeather} />
			</div>
		)
	}
}

WeatherContainer.propTypes = {
	currentWeather: PropTypes.object
}

export const mapStateToProps = state => ({
	currentWeather: state.currentWeather
})

export default connect(mapStateToProps)(WeatherContainer)