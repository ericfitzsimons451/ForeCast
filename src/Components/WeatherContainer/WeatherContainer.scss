.weather-container {
    border: solid 1px;
    height: 100vh;
}