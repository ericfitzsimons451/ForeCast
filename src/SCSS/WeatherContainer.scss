.weather-container {
	height: 100vh;
	display: flex;
	justify-content: center;
	background: url('https://www.chubmagazine.com/wp-content/uploads/2017/10/Greengrass.jpg');
}