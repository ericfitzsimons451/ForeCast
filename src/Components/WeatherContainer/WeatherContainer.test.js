import React from 'react'
import { WeatherContainer, mapStateToProps } from './WeatherContainer'
import { shallow } from 'enzyme'

describe('WeatherContainer', () => {
    it('should match the snapshot', () => {
        const mockWeather = {
            city: 'Denver'
        }
        const wrapper = shallow(<WeatherContainer currentWeather={mockWeather}/>)
        expect(wrapper).toMatchSnapshot()
    })

    describe('mapStateToProps', () => {
        const mockWeather = {
            city: 'Denver'
        }
        const initialState = {}
        const mockState = {
            currentWeather: mockWeather
        }
        expect(initialState).toEqual({})
        const mappedProps = mapStateToProps(mockState)
        expect(mappedProps).toEqual(mockState)
    })
})