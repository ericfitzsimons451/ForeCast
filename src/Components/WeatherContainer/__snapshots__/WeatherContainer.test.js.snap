// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherContainer should match the snapshot 1`] = `
<div
  className="weather-container"
>
  <City
    currentWeather={
      Object {
        "city": "Denver",
      }
    }
    key="23"
  />
</div>
`;
